### DISCUSSION TITLE
1.5.3

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#628] [Обновления]
- Обновлен Yandex HMAC (#627)
- Уменьшена задержка перевода до 30 секунд, если нет субтитров от Яндекса и 20 секунд, если субтитры есть (#621)
- Перевод текста (/translate & /detect) через Yandex API, теперь, работает через GET запросы, благодаря чему задержка была немного уменьшена
- Добавлен перевод возвращаемых сервером ошибок через Deepl
- Исправлено не консистентное отображение установленого значения в дропдаунах "Detect Service" и "Translate errors from the API" (сохраненное значение было написано маленькими буквами, а только что установленное большими)
- Добавлен фалбек, если есть ссылка на видео, но нет ссылки на субитры, для Udemy и Coursera
- Фикс неправильного отображения кнопки в некоторых случаях (если выбрана нестандартная позиция кнопки) (дофикс)
- Исправлен баг при переключение видео на Rutube
- Добавлена доп. проверка для GM_fetch
- Функциионал очистки вынесен в отдельную функцию cleanText
- Другие мелкие фиксы и оптимизация кода (#603, #612, #616, #621, #623)

## What's Changed
* Исправление fetch при использовании AdGuard by @thedanill in https://github.com/ilyhalight/voice-over-translation/pull/594
* mini patch №2 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/603
* Перенос обработчика в функцию addExtraEventListener by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/612
* Замена progress на autoTranslate by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/616
* Оптимизация кода и уменьшение задержки перевода by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/618
* timeoutDuration by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/621
* Замена forEach на for и небольшие исправления by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/623
* Update HMAC key by @thedanill in https://github.com/ilyhalight/voice-over-translation/pull/627

## New Contributors
* @thedanill made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/594

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.5.2.2...1.5.3

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.3'>1.5.3</a>.</em>

**Comment by ilya-corneli** [comment]
Спасибо большое!!!

**Comment by Tynch05** [comment]
Спасибо за релиз. У меня такой вопрос.
Кнопка "Переводить при открытии" работает для всех сайтов.
А на старой версии работала только для того сайта, где включена галочка.
Например в ВК включу "переводить при открытии", а в ютубе нет.
Я сидел на старой версии 1.4.1. Но недавно эта версия сдохла. Теперь там только "Не удалось загрузить перевод".
![image](https://github.com/ilyhalight/voice-over-translation/assets/33279455/eb4fd9e5-f7e5-4802-8da1-6632df7d94f4)

**Reply by ilyhalight** [reply]
Сейчас настройки между всему сайтами общие, поэтому галочка всегда синхронизируется. Может быть в будущем сделаю, чтобы можно было некоторые настройки изменять отдельно (но это не точно)

**Reply by FiLinX** [reply]
а скажите, как же всё таки 1.4.1 версия работала до позавчера? хотя в ней нету строки с хмаккеем)
я тоже на одном из компов эту версию пользовал..

**Reply by Tynch05** [reply]
> а скажите, как же всё таки 1.4.1 версия работала до позавчера? хотя в ней нету строки с хмаккеем) я тоже на одном из компов эту версию пользовал..

Я не знаю, как он работал. Давно устанавливал код 1.4.1 как скрипт в tampermonkey. И с тех пор не трогал, пока недавно не перестал работать
![image](https://github.com/ilyhalight/voice-over-translation/assets/33279455/6551f4ce-912e-416b-9576-1f8bfc803d55)

Но теперь 1.4.1 на любом сайте не может загрузить перевод
![image](https://github.com/ilyhalight/voice-over-translation/assets/33279455/4c5468eb-aa7b-4a0d-a646-5be7e45a6c77)

[ilyhalight](https://github.com/ilyhalight)
У меня еще вопрос, а можно ли как-нибудь что-то скопировать и заменить из новых версий на старую?
Чтобы старая работал или нет? Я просто тупой и в программировании 0. Если направите изучать JS или еще что-то, то попробую изучить

**Reply by FiLinX** [reply]
ой, вот я дурень то)) 
я таки нашёл в той 1.4.1 строку с хмаккеем!
я то думал шо браузерным поиском по Ctrl+F всё видит на странице редактора скрипта в темпермоке
а оказалось что нифига это не так))  и искать надо через темпермонковскую Найти
уф..

**Reply by FiLinX** [reply]
а можно ли как-нибудь что-то скопировать и заменить из новых версий на старую? Чтобы старая работал или нет?

[Tynch05](https://github.com/Tynch05) - зайди в темпермонку, кликни по стоящему 1.4.1, откроется редактор скрипта, нажми в его менюшке Найти-Найти, и в появившейся внизу окна форме, вбей -  xtGCyGdTY2Jy6OMEKdTuXev3Twhkamgm
и замени найденное на это  -  bt8xH3VOlb4mqf0nqAibnDOoiPlXsisf
затем жми в меню Файл на Сохранить

**Reply by ilyhalight** [reply]
> У меня еще вопрос, а можно ли как-нибудь что-то скопировать и заменить из новых версий на старую?

Можно, но заниматься этим нет смысла

**Reply by Tynch05** [reply]
> Можно, но заниматься этим нет смысла
Ок, спасибо


> зайди в темпермонку, кликни по стоящему 1.4.1, откроется редактор скрипта, нажми в его менюшке Найти-Найти, и в появившейся внизу окна форме, вбей - xtGCyGdTY2Jy6OMEKdTuXev3Twhkamgm
> и замени найденное на это - bt8xH3VOlb4mqf0nqAibnDOoiPlXsisf
> затем жми в меню Файл на Сохранить
Братела спасибо большое. Сделал, всё заработало. Теперь понял, что за ключи и хмаккей
